src/python/pants/backend/python/goals/export.py

```diff
@@ -143,15 +144,20 @@ async def export_virtualenv(
                     "--remove=all",
```

Comment:
This is a busted cherry-pick. 2.14.x goes from `rm -rf ~/.pex` to `rm -rf ~/.cache/pants/named_caches/pex_root` since the `--remove=all` -> `--remove=pex` bit of the CP was missed.
This explains all of:
+ https://github.com/pantsbuild/pex/issues/1955
+ https://github.com/pantsbuild/pex/issues/1956
+ https://github.com/pantsbuild/pex/issues/1957

Reply:
I'm working up a patch to right this...

Reply:
https://github.com/pantsbuild/pants/pull/17312

Reply:
Oooof, so I left off the most important bit. Ouch... 🤦 🤦 🤦 🤦 🤦